modules/top-level/files/default.nix

```diff
@@ -59,12 +60,37 @@ in
       warnings = concatFilesOption "warnings";
       assertions = concatFilesOption "assertions";
 
+      # Add files to extraFiles
+      extraFiles = lib.mkDerivedConfig options.files (
+        lib.mapAttrs' (
+          _: file: {
+            name = file.path;
+            value.source = file.plugin;
+          }
+        )
+      );
+
       # A directory with all the files in it
-      filesPlugin = pkgs.buildEnv {
-        name = "nixvim-config";
-        paths =
-          (lib.mapAttrsToList (_: file: file.plugin) files)
-          ++ (lib.mapAttrsToList pkgs.writeTextDir config.extraFiles);
-      };
+      filesPlugin = pkgs.runCommandLocal "nvim-config" { } ''
+        set -euo pipefail
+
+        makeEntry() {
+          src="$1"
+          target="$2"
+          mkdir -p "$out/$(dirname "$target")"
+          cp "$src" "$out/$target"
+        }
+
+        mkdir -p "$out"
+        ${lib.concatMapStringsSep "\n" (
+          { target, source, ... }:
+          lib.escapeShellArgs [
+            "makeEntry"
+            # Force local source paths to be added to the store
+            "${source}"
+            target
+          ]
+        ) extraFiles}
+      '';
```

Comment:
This implementation is taken from here: https://github.com/NixOS/nixpkgs/blob/7dca15289a1c2990efbe4680f0923ce14139b042/nixos/modules/system/etc/etc.nix#L275-L293

Should probably have a reference/attribution comment?